launch/launch/frontend/parse_substitution.py

```diff
@@ -93,14 +93,21 @@ def template(self, fragments):
     double_quoted_template = template
 
 
-grammar_file = os.path.join(get_package_share_directory('launch'), 'frontend', 'grammar.lark')
-parser = Lark.open(grammar_file, start='template')
-transformer = ExtractSubstitution()
+def get_grammar_path():
+    return os.path.join(
+        get_package_share_directory('launch'), 'frontend', 'grammar.lark')
+
+
+_parser = None
```

Comment:
In general, thumbs up to removing stuff from import time.  I guess my question is: does this have to be global at all?  Can't we just instantiate the Lark parser during `parse_substitution` below when we need it?

Reply:
It's possible, but we would initialize it many times, which is not needed.
Using a global looks good to me.

Reply:
The problem with the global is that we're never closing the file.
Maybe using a local isn't bad.

Reply:
Parsing a grammar is a pretty heavy process. So doing it more than one isn't a good idea imo. That is why delaying the steps until it is needed is also beneficial to not affect the module load time.